dpctl-capi/source/dpctl_sycl_context_interface.cpp

```diff
@@ -116,6 +116,36 @@ DPCTLContext_Copy(__dpctl_keep const DPCTLSyclContextRef CRef)
     }
 }
 
+__dpctl_give DPCTLDeviceVectorRef
+DPCTLContext_GetDevices(__dpctl_keep const DPCTLSyclContextRef CRef)
+{
+    auto Context = unwrap(CRef);
+    if (!Context) {
+	std::cerr << "Can not retrieve devices from DPCTLSyclContextRef as input is a nullptr\n";
+	return nullptr;
+    }
+    vector_class<DPCTLSyclDeviceRef> *DevicesVectorPtr = nullptr;
+    try {
+        DevicesVectorPtr = new vector_class<DPCTLSyclDeviceRef>();
+    } catch (std::bad_alloc const &ba) {
+        // \todo log error
+        std::cerr << ba.what() << '\n';
+        return nullptr;
+    }
+    try {
+	auto Devices = Context->get_devices();
```

Comment:
Do we need a `runtime_error` check here? Can `get_devices()` throw?

Reply:
I can add that, although it appears as if context is just returning something it stores internally